aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -525,4 +531,116 @@ public AppSyncStore getStore() {
+     *
+     * @param baseQuery the base query to get the baseline state
+     * @param baseQueryCallback callback to handle the baseQuery results
+     * @param baseRefreshIntervalInSeconds time duration (specified in seconds) when the base query will be re-run to get an updated baseline state.
+     * @param <D>
+     * @param <T>
+     * @param <V>
+     * @return a syncId that can be used later to cancel the sync operation.
+     */
+
+    public <D extends Query.Data, T, V extends Query.Variables> Long sync(
+            @Nonnull Query<D, T, V> baseQuery,
+            GraphQLCall.Callback<Query.Data> baseQueryCallback,
+            long baseRefreshIntervalInSeconds) {
+
+        return this.sync(baseQuery, baseQueryCallback,null, null,null, null, baseRefreshIntervalInSeconds);
+    }
+
+
+    /**
+     * Provides the ability to sync using a baseQuery, deltaQuery, and a refresh interval
+     * @param baseQuery the base query to get the baseline state
+     * @param baseQueryCallback callback to handle the baseQuery results
+     * @param deltaQuery the catchup query
+     * @param deltaQueryCallback callback to handle the deltaQuery results
+     * @param baseRefreshIntervalInSeconds time duration (specified in seconds) when the base query will be re-run to get an updated baseline state.
+     * @param <D>
+     * @param <T>
+     * @param <V>
+     * @return a syncId that can be used later to cancel the sync operation.
+     */
+    public <D extends Query.Data, T, V extends Query.Variables> Long sync(
+            @Nonnull Query<D, T, V> baseQuery,
+            GraphQLCall.Callback<Query.Data> baseQueryCallback,
+            Query<D,T,V> deltaQuery,
+            GraphQLCall.Callback<Query.Data> deltaQueryCallback,
+            long baseRefreshIntervalInSeconds) {
+        return this.sync(baseQuery, baseQueryCallback, null, null, deltaQuery, deltaQueryCallback, baseRefreshIntervalInSeconds);
+    }
+
+
+    /**
+     * Provides the ability to sync using a baseQuery and Subscription
+     * @param baseQuery the base query to get the baseline state
+     * @param baseQueryCallback callback to handle the baseQuery results
+     * @param subscription subscription to get changes on the fly
+     * @param subscriptionCallback callback to handle the subscription messages
+     * @param <D>
+     * @param <T>
+     * @param <V>
+     * @return a syncId that can be used later to cancel the sync operation.
+     */
+    public <D extends Query.Data, T, V extends Query.Variables> Long sync(
+            @Nonnull Query<D, T, V> baseQuery,
+            GraphQLCall.Callback<Query.Data> baseQueryCallback,
+            Subscription<D,T,V> subscription,
+            AppSyncSubscriptionCall.Callback subscriptionCallback
+            ) {
+        return this.sync(baseQuery, baseQueryCallback, subscription, subscriptionCallback, null, null, 0 );
+    }
+
+    public void cancelSync(Long id) {
```

Comment:
What is the motivation to move away from the watcher logic of calling `cancel()` to instead use a `cancel(id)`?

Reply:
wasn't aware of the convention here. Will move to using an object that implements the java.util.concurrent.Cancellable interface.